dvc/types.py

```diff
@@ -0,0 +1,10 @@
+from typing import TYPE_CHECKING, Union
+
+from dvc.path_info import PathInfo, URLInfo
+
+if TYPE_CHECKING:
+    from os import PathLike
+
+StrPath = Union[str, "PathLike[str]"]
+DvcPath = Union[PathInfo, URLInfo]
```

Comment:
I am not sure what to call this union. Suggestions would be appreciated.

Reply:
@skshetry I guess in an ideal world we would've made URLInfo inherit from PurePath. Could be done.

Reply:
We should do this soon, as we cannot type check our tree function without this.